tests/unit/CoreQuoteTest.php

```diff
@@ -0,0 +1,245 @@
+	 *   - 'gradient'
+	 *   - 'lock'
+	 *   - 'style'
+	 *   - 'textColor'
+	 *
+	 * @return void
+	 */
+	public function test_retrieve_core_quote_attributes() {
+		$block_content = '
+			<!-- wp:quote {"lock":{"move":true,"remove":true},"fontFamily":"body","fontSize":"small","backgroundColor":"pale-cyan-blue","style":{"elements":{"heading":{"color":{"text":"var:preset|color|vivid-cyan-blue","background":"var:preset|color|cyan-bluish-gray"}}}},"textColor":"vivid-red","gradient":"pale-ocean"} -->
+			<blockquote class="wp-block-quote" id="test-anchor"><!-- wp:heading -->
+			<h2 class="wp-block-heading">Quote, with heading color</h2>
+			<!-- /wp:heading --><cite>Citation</cite></blockquote>
+			<!-- /wp:quote -->
+		';
+
+		// Update the post content with the block content.
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$query     = $this->query();
+		$variables = [
+			'id' => $this->post_id,
+		];
+
+		$actual = graphql( compact( 'query', 'variables' ) );
+
+		$this->assertArrayNotHasKey( 'errors', $actual, 'There should not be any errors' );
+		$this->assertArrayHasKey( 'data', $actual, 'The data key should be present' );
+		$this->assertArrayHasKey( 'post', $actual['data'], 'The post key should be present' );
+		$this->assertEquals( $this->post_id, $actual['data']['post']['databaseId'], 'The post ID should match' );
+
+		// Verify the block data.
+		$block = $actual['data']['post']['editorBlocks'][0];
+		$this->assertNotEmpty( $block['apiVersion'], 'The apiVersion should be present' );
+		$this->assertEquals( 'text', $block['blockEditorCategoryName'], 'The blockEditorCategoryName should be text' );
+		$this->assertNotEmpty( $block['clientId'], 'The clientId should be present' );
+		$this->assertNotEmpty( $block['innerBlocks'], 'There should be no inner blocks' );
+		$this->assertEquals( 'core/quote', $block['name'], 'The block name should be core/quote' );
+		$this->assertEmpty( $block['parentClientId'], 'There should be no parentClientId' );
+		$this->assertNotEmpty( $block['renderedHtml'], 'The renderedHtml should be present' );
+
+		$style = wp_json_encode(
+			[
+				'elements' => [
+					'heading' => [
+						'color' => [
+							'text'       => 'var:preset|color|vivid-cyan-blue',
+							'background' => 'var:preset|color|cyan-bluish-gray',
+						],
+					],
+				],
+			]
+		);
+
+		// Verify the attributes.
+		$this->assertEquals(
+			[
+				'anchor'          => 'test-anchor',
+				'backgroundColor' => 'pale-cyan-blue',
+				'citation'        => 'Citation',
+				'className'       => null,
+				'cssClassName'    => 'wp-block-quote',
+				'fontFamily'      => 'body',
+				'fontSize'        => 'small',
+				'gradient'        => 'pale-ocean',
+				'lock'            => wp_json_encode(
+					[
+						'move'   => true,
+						'remove' => true,
+					]
+				),
+				'style'           => $style,
+				'textColor'       => 'vivid-red',
+				'value'           => '',
```

Comment:
Since we're using a nested Core Heading and `value` is null, we should assert the basics (the inner block is there and it has the correct content).

Reply:
Fixed ✅